index.js

```diff
@@ -1,12 +1,14 @@
 'use strict';
-
+const fs = require('fs');
 /** @class
 * @param {znHttp} znHttp A ZnHttp reference
 */
 function BackendFactory(znHttp) {
 
 	if (!znHttp) {
-		znHttp = require('../../../lib/zn-http')();
+		znHttp = fs.existsSync('../../../lib/zn-http2') ?
+			require('../../../lib/zn-http2')() :
```

Comment:
znHttp2 can't be initialized like this, that's why we couldn't just rewrite znHttp

Reply:
ok then leave it the way it is then? since znHttp will be passed in from eventData ?

Reply:
Yeah, I think this has to stay the way it was so that it's non-breaking, but in use, we should be passing eventData.znHttp into BackendFactory.

Reply:
ok ill close this one then.